maat-court-data-api/src/main/java/gov/uk/courtdata/users/service/UserSummaryService.java

```diff
@@ -55,4 +58,33 @@ public UserSummaryDTO getUserSummary(String username) {
         String currentUserSession = userEntity.isEmpty() ? null : userEntity.get().getCurrentSession();
         return userSummaryMapper.userToUserSummaryDTO(username, newWorkReasonForUser, userRoleActions, reservationsDTO, currentUserSession, roleDataItemDTOList);
     }
+
+    public UserEntity getUser(String username) {
+        Optional<UserEntity> userEntity = userRepository.findById(username);
+        if (userEntity.isEmpty()) {
+            String message = String.format("No User found with user name: [%s]", username);
+            throw new RequestedObjectNotFoundException(message);
+        }
+        return userEntity.get();
+    }
+
+    public void createUser(UserEntity userEntity) {
+        userRepository.saveAndFlush(userEntity);
+    }
+
+    public void deleteUser(String username) {
+        userRepository.deleteById(username);
+    }
+
+    public void patchUser(String username, Map<String, Object> updateFields) {
+        UserEntity userEntity = getUser(username);
+        updateFields.forEach((key, value) -> {
+            Field field = ReflectionUtils.findField(UserEntity.class, key);
```

Comment:
As there is less attributes in the UserEntity, I prefer put over PATCH

Reply:
Keeping it consistent with the recent changes, like Reservations that is used only for the automated tests.